src/privileged/FirefoxMonitor.jsm

```diff
@@ -170,45 +175,34 @@ this.FirefoxMonitor = {
     this.strings = Services.strings.createBundle(`data:text/plain;base64,${b64}`);
   },
 
-  _loadBreachesTimer: null,
-  _breachesLastModified: 0,
+  kRemoteSettingsKey: "fxmonitor-breaches",
   async loadBreaches() {
-    let response;
-    try {
-      response = await fetch(this.breachListURL, {
-        credentials: "omit",
-        headers: {
-          "If-Modified-Since": this._breachesLastModified,
-        },
+    const { RemoteSettings } = ChromeUtils.import("resource://services-settings/remote-settings.js", {});
+
+    let populateSites = (data) => {
+      this.domainMap.clear();
+      data.forEach(site => {
+        this.domainMap.set(site.Domain, {
+          Name: site.Name,
+          PwnCount: site.PwnCount,
+          Year: (new Date(site.BreachDate)).getFullYear(),
+        });
       });
-    } catch (e) {
-      // Fetch only rejects on network failures or other anomalies;
-      // response will be undefined and we'll return early.
     }
 
-    // Arm the refresh timer already, since we may return early if we 304'd.
-    // Commented out for now since we are packaging breaches in the addon.
-    // this._loadBreachesTimer = setTimeout(() => this.loadBreaches(), this.breachRefreshTimeout);
+    RemoteSettings(this.kRemoteSettingsKey).on("sync", (event) => {
+      const { data: { current } } = event;
+      populateSites(current);
+    });
 
-    // If the list hasn't been updated since we last checked, the server
-    // will send a 304 response. In any case, we don't handle anything
-    // except a 200 OK.
-    if (!response || response.status !== 200) {
-      return;
+    const data = await RemoteSettings(this.kRemoteSettingsKey).get();
+    if (data && data.length) {
+      populateSites(data);
+    }
+    else if (this.debug) {
+      // Force a sync if we're debugging. This will trigger the on("sync") callback.
+      RemoteSettings(this.kRemoteSettingsKey).maybeSync(Infinity, Date.now());
```

Comment:
If `this.debug` is the context in which the `RemoteSettings.on("sync")` handler is called, should we wrap it with `if (this.debug) {` too?

Reply:
The on("sync") listener is needed to handle future syncs after we have already loaded breaches.
Calling get() does not trigger a sync. It only returns data that is already available - hence the null-check before calling populateSites.
Upon Firefox launch with a new profile, when this function is called, it's highly likely that the fxmonitor-breaches key has not been sync'd - and get() will not force that. I added the debug flag to force a sync in this case so that QA can test it properly.

Reply:
Makes sense. Thanks for the clarification.